packages/backend-core/src/middleware/authenticated.js

```diff
@@ -128,6 +128,8 @@ module.exports = (
       }
       if (!user && tenantId) {
         user = { tenantId }
+      } else {
+        delete user.password
```

Comment:
looks like this is the second time we're doing this in this function. In the first case we delete when
- there is a session
- there is a matched user

then we proceed to populate the user again if there is an api key and we're not already authenticated, the default public api flow. So is that the case we're covering here? I do like where you've added this near the end, I think we could even remove the other usage as this should over all cases

Reply:
Yes, the second delete was intended to cover users verified via an API key. As you point out, the second one would actually cover both auth flows.